### ISSUE TITLE
PFFile set to nil remains as PFDeleteOperation

### ISSUE THREAD

**Issue by benpackard** [ParsePlatform/Parse-SDK-iOS-OSX#996]
Given a PFObject subclass `Place`, with a `PFFile?` var `photoImage`, the `photoImage` of an instance of `Place` is removed:

```
place.photoFile = nil
self?.place.saveInBackgroundWithBlock() { (_, _) in
    print(place.photoFile)
}
```

Expected: console logs `nil`
Actual: console logs `Optional(delete)`

The `photoFile` is in fact empty when viewed in the Parse Dashboard, but as best as I can tell, the `photoFile` on the `Place` in memory is a `PFDeleteOperation`. Given that execution waited for the deletion to save, this is unexpected and problematic. Imagine a `PFImageView` that is loaded based on the presence of the `photoImage`:

```
if let photoFile = place.photoFile {
    cell.photoImageView.file = photoFile
    cell.photoImageView.loadInBackground()
    }
```

This is no longer safe. Since the deleted `photoImage` is not the expected `nil`, this code crashes attempting to load it:

`-[PFDeleteOperation url]: unrecognized selector sent to instance 0x7ffaab79c060`

This issue can be worked around by setting the image to nil a second time after saving:

```
place.photoFile = nil
self?.place.saveInBackgroundWithBlock() { (_, _) in
    print(place.photoFile)
    place.photoFile = nil
}
```

Expected: console logs `nil`
Actual: console logs `nil`

**Comment by benpackard** [comment]
Interestingly, this issue does not occur if replacing this line from the PFSubclassing file:

`@NSManaged var photoFile: PFFile?`

with:
```
	var photoFile: PFFile? {
		
		get {
			if let photoFile = self["photoFile"] as? PFFile {
				return photoFile
			}
			return nil
		}
		set {
			self["photoFile"] = newValue ?? NSNull()
		}
	}
```

**Comment by benpackard** [comment]
This was fixed by updating parse-server from 2.2.12 to 2.2.17.
